### DISCUSSION TITLE
Markdown math not working properly

### DISCUSSION THREAD

**Discussion by matsutakehoyo** [community/community#19953] [Projects and Issues]
The [newly introduced math support for markdown](https://github.blog/2022-05-19-math-support-in-markdown/) is producing some unexpected results.

Even the example math code does not render correctly.

```
When $a \ne 0$, there are two solutions to $(ax^2 + bx + c = 0)$ and they are 
$$ x = {-b \pm \sqrt{b^2-4ac} \over 2a} $$
```
In my README the second line seems to work fine but the first line results in this

```
When a≠0(ax^2 + bx + c = 0)$ and they are
```

I’ve tried a couple of things. Single inline math seems to work just fine, but when there are multiple inline math expressions the stuff between the end of the first expression and the beginning of the second one get ignored.

**Comment by Banyc** [comment]
The same here. I post an issue at a repo with some math notations [here](https://github.com/pim-book/programmers-introduction-to-mathematics/issues/52), but the notations look weird and not consist with the preview in my local VSCode.

**Comment by manasesjesus** [comment]
You need to use `$` as delimiter for inline text+math, and `$$` will render in a new line. 
If you use a code block ` ```math `, then you don't need the delimiters, but everything inside the block will be treated as mathematics.

**without a code block**
```
When $a \ne 0$, there are two solutions to $ax^2 + bx + c = 0$ and they are:
$$x = {-b \pm \sqrt{b^2-4ac} \over 2a}.$$

_Square root of 9_
$$\sqrt{9}$$
```

**with code blocks**

```
    Some math here:
    ```math
    x = {-b \pm \sqrt{b^2-4ac} \over 2a} 
    ```
    and here:
    ```math
    \sqrt{3}
    ```
```
will be displayed as follows:


<img width="533" alt="Captura de pantalla 2022-07-04 a la(s) 2 23 44 pm" src="https://user-images.githubusercontent.com/24204142/177154643-21037cb9-8dd7-461b-b771-7fba136ed03d.png">

**Reply by matsutakehoyo** [reply]
@manasesjesus just to be clear I only use ``` to illustrate. The behavior I'm getting is this. I didn't know about the math block though. 

I type this 
```
When $a \ne 0$, there are two solutions to $(ax^2 + bx + c = 0)$ and they are 
$$ x = {-b \pm \sqrt{b^2-4ac} \over 2a} $$
```

and get this

When $a \ne 0$, there are two solutions to $(ax^2 + bx + c = 0)$ and they are 
$$ x = {-b \pm \sqrt{b^2-4ac} \over 2a} $$

note the second inline math expression is not parsed correctly

**Reply by manasesjesus** [reply]
@matsutakehoyo if I copy & paste your example in this reply, I get the same as you said:

When $a \ne 0$, there are two solutions to $(ax^2 + bx + c = 0)$ and they are 
$$ x = {-b \pm \sqrt{b^2-4ac} \over 2a} $$

if I copy, paste & remove the parenthesis, it works fine:

When $a \ne 0$, there are two solutions to $ax^2 + bx + c = 0$ and they are 
$$ x = {-b \pm \sqrt{b^2-4ac} \over 2a} $$

The parenthesis have no effect in that operation, and it seems that dollar+parenthesis `$(` is the issue. 

It's possible to use parenthesis in inline math when it makes sense, like this one $(x+2)/(y+3)$ and this one $9*(ax^2 + 1) + bx + c = 0$. More complex stuff should go in a block, like this one: 

$$\left( \sum_{k=1}^n a_k b_k \right)^2 \leq \left( \sum_{k=1}^n a_k^2 \right) \left( \sum_{k=1}^n b_k^2 \right)$$

**Reply by matsutakehoyo** [reply]
@manasesjesus Interesting… eliminating the parenthesis does fix the issue with the example.

though I don't think the parenthesis is the only issue. I need to write a lot of math inside sentences, and currently almost none of it is working properly. For example, I have this sentence

```
The outcome of the i-th trial $y_i$, with success = 1 or fail = 0, can be modeled with a Bernoulli distribution with probability to answer correctly $\theta_i$.
```

this doesn't have parenthesis, but still fails

The outcome of the i-th trial $y_i$, with success = 1 or fail = 0, can be modeled with a Bernoulli distribution with probability to answer correctly $\theta_i$.

**Reply by manasesjesus** [reply]
**interesting... seems like an issue with the math library implementation. The following:**
```
$y_i$ and $\theta_i$ at the beginning
$\sqrt{3x-1}+(1+x)^2$ at the end $\theta_i$ and $y_i$
in the middle $\theta_i$ and $y_i$ more text here
``` 
**Works fine:**
$y_i$ and $\theta_i$ at the beginning
$\sqrt{3x-1}+(1+x)^2$ at the end $\theta_i$ and $y_i$
in the middle $\theta_i$ and $y_i$ more text here

**but if we take the third line and invert the order, it fails:**
in the middle $y_i$ and $\theta_i$ more text here

**it also fails when using multiple times the** `$\`.
multiple $\theta_i$ times $\theta_i$ in a line $\theta_i$ 

**it seems that also fails in long lines, but if I split your example in two lines, it works:** 
The outcome of the i-th trial $y_i$, with success = 1 or fail = 0, can be modeled with 
a Bernoulli distribution with probability to answer correctly $\theta_i$.

**You may try different lines or order for it to work. Meanwhile, go ahead and report the issue to GitHub with your examples.**

**Reply by manasesjesus** [reply]
@matsutakehoyo I just checked and you're not alone 😄. Many others have already reported the issue. My guess is, GitHub will eventually fix it. Meanwhile, may the force be with you.

**Reply by matsutakehoyo** [reply]
@manasesjesus Thanks for your comments! Just for future references? What is the best place to report this issue, I couldn’t  find a specific repo for this and ended up here…

**Reply by manasesjesus** [reply]
Here are some resources @matsutakehoyo :
- [GitHub Support - Bug Report](https://support.github.com/contact/bug-report)
- [GitHub Support](https://support.github.com/request)
- [GitHub Community Forum](https://github.community/c/github-help/48)

**Reply by manasesjesus** [reply]
@Banyc FYI 
you may also report your bugs.

**Reply by manasesjesus** [reply]
@matsutakehoyo @Banyc there was an update. You may check if your issues are solved now.

**Comment by nschloe** [comment]
This appears to be fixed now:

---
When $a \ne 0$, there are two solutions to $(ax^2 + bx + c = 0)$ and they are 

$$
x = {-b \pm \sqrt{b^2-4ac} \over 2a}
$$

---

**Comment by yasirroni** [comment]
The `($` issue is still not fixed, thus I can't write `($P_{g}$)`:
____
($P_{g}$)
___

**Reply by yasirroni** [reply]
It seems it is fixed now (?)

**Comment by masagrator** [comment]
It seems now this doesn't work if you use markdown > 

```
> What is Skin temperature ($\mathbf{T}skin$)?
```

Example
> What is Skin temperature ($\mathbf{T}skin$)?

It worked before.

Edit: It was fixed.

**Comment by gpanda** [comment]
Try inline math $\sqrt{x}$

**Comment by holderbp** [comment]
I can't get double-dollar-sign math blocks to work.  The math code block method given above does work, though.  Is double dollar sign broken for some reason just for me?

See my README.md file displayed here: [https://github.com/holderbp/lecture_notes/tree/main](https://github.com/holderbp/lecture_notes/tree/main)

**Reply by nschloe** [reply]
> Is double dollar sign broken for some reason just for me?

It's super broken. Always use `math` code blocks.

**Reply by TechProofreader** [reply]
There still seem to be issues across the board with math markdown on GitHub. For me, I can only use `$` or `$$` because `math` code blocks aren't working. It seems hit or miss, so part of me wonders if there are conflicts between how GitHub or MathJax is implementing the code vs how certain browsers are drawing the math code.

Another interesting issue is that I'm finding if you create a large square root symbol, `$$` causes the vertical bar of the symbol to get cut off unless you zoom in, whereas `$` formats it properly no matter what. I actually just experienced this while updating the docs for a Microsoft repository, ironically.

**Comment by simonary** [comment]
Inline math seems not working with non-english text?

```text
For every integer $i$
```

For every integer $i$

```text
对任意整数$i$
```

对任意整数$i$

**Reply by gpanda** [reply]
对任意整数 $i$
For every integer$i$

You need at least a space between inline math and normal text.

**Comment by kyle-tennison** [comment]
Having issues with tagged equations...

$$E \equiv \sum_i \dot q_i \frac{\partial L}{\partial \dot q_i} -L \tag{1}$$

Screenshot in case it's specifically my problem...
<img width="757" alt="image" src="https://github.com/user-attachments/assets/d0c985e7-2374-4208-a858-70330348c7fd" />

**Reply by kyle-tennison** [reply]
Using `$$E \equiv \sum_i \dot q_i \frac{\partial L}{\partial \dot q_i} -L \tag{1}$$`; the `\tag{_}` will throw anything into this weird vertical jumble.

**Reply by rodrigo-cl-porto** [reply]
The only working way I've found out is by using arrays, like:

```
$$
\begin{array}{lr}
    a^2 + b^2 = c^2 & \text{(1)}
\end{array}
$$
```

$$
\begin{array}{lr}
    a^2 + b^2 = c^2 & \text{(1)}
\end{array}
$$

You can add an `<span>` tag to add an reference to the equation (make sure to have line feeds right after `<span>` and right before the `</span>`:

```
<span id="eq2">

$$
\begin{array}{lr}
    a^2 + b^2 = c^2 & \text{(2)}
\end{array}
$$

</span>
```

<span id="eq2">

$$
\begin{array}{lr}
    e^{i \pi} = -1 & \text{(2)}
\end{array}
$$

</span>

That way, you can add the link to the [Equation (2)](#eq2) by using `[Equation (2)](#eq2)`
